shopinvader_api_signin_jwt/routers/signin.py

```diff
@@ -34,22 +34,12 @@ def signin(
     Authenticate the partner based on a JWT token or a session cookie.
     Set the session cookie if allowed.
     Return HTTP code 201 if res.partner created (case of the first signin).
-    Transfer anonymous cart and delete anonymous partner if any.
     """
     if not partner:
         partner = env[
             "shopinvader_api_signin_jwt.signin_router.helper"
         ]._create_partner_from_payload(payload)
         response.status_code = status.HTTP_201_CREATED
```

Comment:
@sbidoul : good catch for the get that do stuff

@paradoxxxzero 

As the aim is to have a generic flow (independent of the authentication mecanism) that will do action of record related to an anonymous partner, maybe the best it to call here the promote method right here ?


```suggestion
        response.status_code = status.HTTP_201_CREATED
        env["res.partner"]._promote_anonymous_partner(
            partner, request.cookies, response
        )
```


@sbidoul also having a generic method on the partner to manage the promote will give the possibility to manage other promote case (like creating a lead form and then sign in... the idea is that every api that allow anonymous partner should implement the related promote without depending on a specific auth mecanisme)